src/resources/assets/js/geo/components/geoMapModal.vue

```diff
@@ -65,39 +64,40 @@ export default {
     methods: {
         // trigger addRule() on parent
         callback(msg) {
-            if (msg == 'ok') {
-                this.$emit('on');
+            if (msg) {
+                this.$emit('on', this.key);
             } else {
-                return null;
+                this.$emit("close-modal");
             }
         },
         handleSelectedImages(ids) {
             if (ids.length > 0) {
-                sessionStorage.setItem(this.key, JSON.stringify(ids));
+                sessionStorage.setItem(this.key, JSON.stringify(ids.sort()));
+                this.disabled = false;
             } else {
                 sessionStorage.removeItem(this.key);
+                this.disabled = true;
             }
         },
-        getImageFilterApi(id) {
-            return LabelApi.get({vid: this.volumeId, lid: id}, {});
-        },
     },
-    watch: {
+    created() {
+        // check whether preselected images exist
+        if(this.selectedImages.length > 0) {
+            this.disabled = false;
+        }
         // show the modal upon trigger-event
-        trigger: function() {
-            this.startLoading();
-            this.showModal = true;
-            // get all image + coordinate information from volume-images
-            CoordApi.get({id: this.volumeId}, {})
-                .then(
-                    (response) => {
-                        this.images = response.body;
-                        this.finishLoading();
-                },
+        this.startLoading();
+        this.show = true;
+        // get all image + coordinate information from volume-images
+        CoordApi.get({id: this.volumeId}, {})
+            .then(
                 (response) => {
-                    return this.handleErrorResponse(response);
-                });
-        },
+                    this.images = response.body;
+                    this.finishLoading();
+            },
+            (response) => {
+                return this.handleErrorResponse(response);
+            });
```

Comment:
`finishLoading()` should be called in a `finally()`, otherwise loading would not finish on error:

```js
CoordApi.get({id: this.volumeId})
   .then(r => this.images = r.body, this.handleErrorResponse)
   .finally(this.finishLoading);
```